.github/workflows/tests.yml

```diff
@@ -0,0 +1,36 @@
+name: Tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - ready_for_review
+
+env:
+  HATCH_VERSION: "1.14.0"
+  OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 1
+      matrix:
+        python-version: [3.9, 3.10.x, 3.11, 3.12]
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install Hatch
+        run: pip install hatch==${{ env.HATCH_VERSION }}
+
+      - name: Run unit and integration tests
+        run: hatch run test:all
```

Comment:
I would suggest to also run and report coverage.
These commands can serve as examples: https://github.com/deepset-ai/haystack-core-integrations/blob/4f067b9b183020d039be017990eb7f868e116099/integrations/amazon_bedrock/pyproject.toml#L53

Even if the report is only visible in GitHub Actions, it is still inspectable
An example: https://github.com/deepset-ai/haystack-core-integrations/actions/runs/12397435961/job/34607798450?pr=1255